src/microbe_stage/MicrobeAI.cs

```diff
@@ -119,281 +68,121 @@ public void Think(float delta, Random random, MicrobeAICommonData data)
+                SetMoveSpeed(0.0f);
             }
+        }
 
-            case LifeState.FLEEING_STATE:
-            {
-                if (predator == null)
-                {
-                    GetNearestPredatorItem(data.AllMicrobes);
-                }
-
-                // In this state you run from predatory microbes
-                if (predator != null)
-                {
-                    DealWithPredators(random);
-                }
-                else
-                {
-                    if (RollCheck(SpeciesActivity, 400, random))
-                    {
-                        lifeState = LifeState.PLANTLIKE_STATE;
-                        boredom = 0;
-                    }
-                    else
-                    {
-                        lifeState = LifeState.NEUTRAL_STATE;
-                    }
-                }
-
-                break;
-            }
+        // Clear the absorbed compounds for run and rumble
+        microbe.TotalAbsorbedCompounds.Clear();
+    }
 
-            case LifeState.PREDATING_STATE:
-            {
-                // Peg your prey
-                if (!preyPegged)
-                {
-                    prey = null;
-                    prey = GetNearestPreyItem(data.AllMicrobes);
-                    if (prey != null)
-                    {
-                        preyPegged = true;
-                    }
-                }
+    private FloatingChunk GetNearestChunkItem(List<FloatingChunk> allChunks, List<Microbe> allMicrobes, Random random)
+    {
+        FloatingChunk chosenChunk = null;
 
-                if (preyPegged && prey != null)
-                {
-                    DealWithPrey(data.AllMicrobes, random);
-                }
-                else
-                {
-                    if (RollCheck(SpeciesActivity, 400, random))
-                    {
-                        lifeState = LifeState.PLANTLIKE_STATE;
-                        boredom = 0;
-                    }
-                    else
-                    {
-                        lifeState = LifeState.NEUTRAL_STATE;
-                    }
-                }
+        // If the microbe cannot absorb, no need for this
+        if (microbe.Membrane.Type.CellWall)
+        {
+            return null;
+        }
 
-                break;
-            }
+        Vector3 testPosition = new Vector3(0, 0, 0);
+        bool hasTestPosition = true;
 
-            case LifeState.SCAVENGING_STATE:
+        // Retrieve nearest potential chunk
+        foreach (var chunk in allChunks)
+        {
+            if ((SpeciesOpportunism == Constants.MAX_SPECIES_OPPORTUNISM) ||
+                ((microbe.EngulfSize * (SpeciesOpportunism / Constants.OPPORTUNISM_DIVISOR)) >
+                    chunk.Size))
             {
-                if (targetChunk == null)
+                if (chunk.ContainedCompounds.Compounds.Where(x => microbe.Compounds.IsUseful(x.Key)).Any())
```

Comment:
```suggestion
                if (chunk.ContainedCompounds.Compounds.Any(x => microbe.Compounds.IsUseful(x.Key)))
```
isn't that equivalent and shorter code?

Reply:
It is, thank you.